architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,236 @@
+
+### Address Type
+
+The primary configuration consumed by Ztunnel is the [`Address` resource](../../pkg/workloadapi/workload.proto).
+As the name suggests, an `Address` represents a particular IP Address.
+This can be a Service or a Workload.
+
+The address type has the following goals:
+* It should support (but not require) on-demand lookups.
+  * Specifically, ztunnel should be able to send a request to the control plane to answer "I got a request to send traffic to 1.1.1.1, what is 1.1.1.1?"
+  * While this is not needed for small scales, this is important for the long tail of massive clusters (think 1 million endpoints), where the entire set of endpoints cannot reasonably be replicated to each ztunnel.
+* It should not be client-specific.
+  * In Istio sidecars, historically we had a lot of client-specific XDS. For example, putting the XDS-client's IP back into the XDS response. This makes efficient control plane implementation (most notably, caching), extremely challenging.
+  * In practice, this largely means that references are fully qualified in the API. IP Addresses (generally) have a network associated with them, node names have a cluster associated with them, etc.
+
+See the [XDS Evolution](https://docs.google.com/document/d/1V5wkeBHbLSLMzAMbwFlFZNHdZPyUEspG4lHbnB0UaCg/edit) document for more history and details.
+
+The `Workload` aims to represent everything about a workload (generally a `Pod` or `WorkloadEntry`).
+This includes things like it's IP address, identity, metadata (name, namespace, app, version, etc), and whether it has a waypoint proxy associated.
+
+The `Service` aims to represent everything about a service (generally a `Service` or `ServiceEntry`).
+This includes things like its IP Addresses and ports.
+
+### Authorization Type
+
+A secondary configuration consumed by Ztunnel is the [`Authorization` resource](../../pkg/workloadapi/security/authorization.proto).
+[Original Design Doc](https://docs.google.com/document/d/17mRVzXe8PS7VoligvIx52T10tOP7xPQ9noeOzoLO2cY/edit).
+
+This resource aims to represent the relatively small set of Authorization policies that Ztunnel support.
+Most notably, this is only L4 resources.
+
+Most of the API is fairly straight forward.
+However, one interesting aspect is how these policies associate with workloads.
+Istio's AuthorizationPolicy has label selectors.
+However, we intentionally do not send those as part of the Workload API, in order to keep the size low.
+
+The obvious solution to this is to put the list of selected workloads into the policy itself.
+However, this means anytime a workload changes (often), we need to update the policy.
+
+Instead, the opposite was chosen: each workload will list the policies that select it.
+This works out to be more efficient in common cases where policies change must less often than workloads.
+This only applies for selector-based policies; namespaced and global policies can be handled without needing to list them out.
```

Comment:
I think this is correct and may be clearer if "without needing to list them out *in the Workload API*."